src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1218,12 +1216,15 @@ void DoorLockServer::clearYearDayScheduleCommandHandler(chip::app::CommandHandle
     emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
 }
 
-bool DoorLockServer::HasFeature(chip::EndpointId endpointId, DoorLockFeature feature)
+chip::BitFlags<DoorLockFeature> DoorLockServer::GetFeatures(chip::EndpointId endpointId)
 {
-    uint32_t featureMap = 0;
-    bool success        = GetAttribute(endpointId, Attributes::FeatureMap::Id, Attributes::FeatureMap::Get, featureMap);
-
-    return success && ((featureMap & to_underlying(feature)) != 0);
+    chip::BitFlags<DoorLockFeature> featureMap;
+    if (!GetAttribute(endpointId, Attributes::FeatureMap::Id, Attributes::FeatureMap::Get, *featureMap.RawStorage()))
```

Comment:
Why not using:

Attributes::FeatureMap::Get(endpointId, &featureMap);

Reply:
Guessing we would want:

```
Attributes::FeatureMap::Get(endpointId, featureMap.RawStorage());
```

I dislike that we have to use raw storage.

Reply:
@andy31415 The problem is that the featuremap attribute is a global attribute, so its type is just "uint32_t" without reference to the specific cluster bitmap (which may not even exist)....